react/week5/components/SearchUser.js

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useState } from 'react';
+import User from './User';
+
+function SearchUser({ username }) {
+  const [isLoading, setIsloading] = useState(true);
+  const [user, setuser] = useState([]);
+
+  useEffect(() => {
+    fetch(`https://api.github.com/search/users?q=${username}`)
+      .then((response) => response.json())
+      .then((data) => {
+        setIsloading(false);
+        setuser(data.items);
+      });
+  }, [username]);
+  return (
+    <div>
+      {typeof user === 'undefined' ? (
+        <div>
+          Maximum number of login attempts exceeded. Please try again later.
+        </div>
+      ) : isLoading ? (
+        <div>loading..</div>
+      ) : (
+        <div>
```

Comment:
Not a big fan of this. I can feel that i need to spend a bit too much energy keeping track of all the states in my brain. Its a little bit too smart in my opinion. See if you can simplify this a little.